mathematica/integrator_plots.cpp

```diff
@@ -219,137 +328,337 @@ class WorkErrorGraphGenerator {
     }
     CHECK_OK(bundle.Join());
 
-    std::vector<std::string> q_error_data;
-    std::vector<std::string> v_error_data;
-    std::vector<std::string> e_error_data;
+    std::vector<std::vector<std::vector<std::tuple<double, Length>>>>
+        q_error_data;
+    std::vector<std::vector<std::vector<std::tuple<double, Speed>>>>
+        v_error_data;
+    std::vector<std::vector<std::vector<std::tuple<double, Energy>>>>
+        e_error_data;
     std::vector<std::string> names;
+    std::vector<std::vector<int>> orders;
+    std::vector<std::string> types;
     for (int i = 0; i < methods_.size(); ++i) {
-      q_error_data.emplace_back(
-          PlottableDataset(evaluations_[i], q_errors_[i], PreserveUnits));
-      v_error_data.emplace_back(
-          PlottableDataset(evaluations_[i], v_errors_[i], PreserveUnits));
-      e_error_data.emplace_back(
-          PlottableDataset(evaluations_[i], e_errors_[i], PreserveUnits));
-      names.emplace_back(ToMathematica(methods_[i].name));
+      q_error_data.emplace_back();
```

Comment:
`auto& q_error_data_for_method = q_error_data.emplace_back();`

Same everywhere.  All these `back()` make me throw up in my mouth a little bit.

Reply:
Yes, and in Mathematica I end up with things like `eErrorData[[indices, ;; ;; 1, 3 ;;]]` and `eErrorData[[indices, ;; ;; 1, 2 ;;]]`. But I don’t have any better ideas here.